### PR TITLE
Update output API docs + minor doc fixes

### PR DESCRIPTION
**Pull request by setrofim**

### CODE REVIEW

File: doc/source/api/output.rst [hunk 1]
```diff
@@ -0,0 +1,443 @@
...
+running the workload. It would be generated either by the workload or by one of
+the augmentations active during the execution of the workload.
+
+A :class:`Metric` has the following attributes:
+
+name
+    The name of the metric.
+
+    .. note:: A name of the metric is not necessarily unique, even for the same
+              job. Some workload by internally run multiple sub-tests each
```
Comment (marcbonnici) [inline doc/source/api/output.rst]: Don't need "by" here.

File: doc/source/api/output.rst [hunk 2]
```diff
@@ -0,0 +1,443 @@
...
+                    artifacts, where as a network filer archiver may choose
+                    to archive them).
+
+    .. note:: The kind parameter is intended to represent the logical
+              function of a particular artifact, not it's intended means of
+              processing -- this is left entirely up to the output
+              processors.
+
+description
+    The may be used by the artifact's creator to provide additional free-form
```
Comment (marcbonnici) [inline doc/source/api/output.rst]: /r/ "The" -> "This"

File: doc/source/developer_reference/output_processing_example.rst [hunk 3]
```diff
@@ -1,96 +0,0 @@
-
```
Comment (marcbonnici) [inline doc/source/developer_reference/output_processing_example.rst]: Any still relevant information from ``developer_reference/APIs.rst`` should also be moved into the new API directory and the old file deleted with its reference in ``developer_reference.rst`` removed.

Comment (setrofim) [reply doc/source/developer_reference/output_processing_example.rst]: Yup, just pushed another commit that deals with that.

**Comment by setrofim** [comment]
Comments addressed.

**Pull request merged**